src/mailadm/bot.py

```diff
@@ -0,0 +1,121 @@
+import sys
+import sqlite3
+import time
+
+import deltachat
+from deltachat import account_hookimpl
+from mailadm.db import DB, get_db_path
+from mailadm.commands import add_user, add_token, list_tokens
+import os
+from threading import Event
+
+
+class SetupPlugin:
+    def __init__(self, admingrpid):
+        self.member_added = Event()
+        self.admingrpid = admingrpid
+        self.message_sent = Event()
+
+    @account_hookimpl
+    def ac_member_added(self, chat: deltachat.Chat, contact, actor, message):
+        assert chat.num_contacts() == 2
+        if chat.id == self.admingrpid:
+            self.member_added.set()
```

Comment:
i think the whole ac_member_added function should be under "chat.id == self.admingrpid" branch.  If i add the bot-address (for whatever reason) to another group then this should be handled gracefully -- maybe easiest just silently ignore being added (or maybe the bot directly leaves any non-admin group, but not neccessary right now and maybe we have a case where it's useful the bot is part of other groups soon).

Reply:
this kind of automatically happened when I took out the `assert` statement :)